synapse/lib/layer.py

```diff
@@ -2142,45 +2201,140 @@ async def iterNodeEdgesN2(self, buid, verb=None):
             verb = lkey[32:].decode()
             yield verb, s_common.ehex(n1buid)
 
-    async def iterPropRows(self, form, prop):
+    async def iterFormRows(self, form, stortype=None, startvalu=None):
+        try:
+            indxby = IndxByForm(self, form)
+
+        except s_exc.NoSuchAbrv:
+            return
 
+        async for item in self._iterRows(indxby, stortype=stortype, startvalu=startvalu):
+            yield item
+
+    async def iterPropRows(self, form, prop, stortype=None, startvalu=None):
         try:
-            abrv = self.getPropAbrv(form, prop)
+            indxby = IndxByProp(self, form, prop)
 
         except s_exc.NoSuchAbrv:
             return
 
-        for _, buid in self.layrslab.scanByPref(abrv, db=self.byprop):
+        async for item in self._iterRows(indxby, stortype=stortype, startvalu=startvalu):
+            yield item
 
-            sode = self._getStorNode(buid)
+    async def iterUnivRows(self, prop, stortype=None, startvalu=None):
```

Comment:
What is the upgrade path for a cluster when layer APIs like this change? This one being purely additive probably makes it ok, but these APIs are going to need to be stable similar to the public APIs.